src/LcdMenu.h

```diff
@@ -1,218 +1,244 @@
+#include <LiquidCrystal.h>
+#else
+#include <LiquidCrystal_I2C.h>
+#endif
+
+#include "MenuItem.h"
+
+/**
+ * the LcdMenu class
+ */
+class LcdMenu {
+   private:
+    /**
+     * Cursor position
+     */
+    uint8_t cursorPosition = 1;
+    /**
+     * First visible item's position in the menu array
+     */
+    uint8_t top = 1;
+    /**
+     * Last visible item's position in the menu array
+     */
+    uint8_t bottom;
+    /**
+     * Rows on the LCD Display
+     */
+    uint8_t maxRows;
+    /**
+     * Columns on the LCD Display
+     */
+    uint8_t maxCols;
+    /**
+     * Array of menu items
+     */
+    MenuItem* currentMenuTable;
+    /**
+     * Down arrow (↓)
+     */
+    byte downArrow[8] = {
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b10101,  // * * *
+        0b01110,  //  ***
+        0b00100   //   *
+    };
+    /**
+     * Up arrow (↑)
+     */
+    byte upArrow[8] = {
+        0b00100,  //   *
+        0b01110,  //  ***
+        0b10101,  // * * *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100   //   *
+    };
+    /**
+     * Draws the cursor
+     */
+    void drawCursor();
+    /**
+     * Draw the menu items with up and down indicators
+     */
+    void drawMenu();
+    /**
+     * Check if the cursor is at the start of the menu items
+     *
+     * @return true : `boolean` if it is at the start
+     */
+    boolean isAtTheStart();
+    /**
+     * Check if the cursor is at the end of the menu items
+     *
+     * @return true : `boolean` if it is at the end
+     */
+    boolean isAtTheEnd();
+    /*
+     * Draw the menu items and cursor
+     */
+    void paint();
+    /**
+     * Reset the display
+     */
+    void reset();
+    /**
+     * Places the cursor at end of Menu's text.
+     *
+     * @param item MenuItem where the cursor should be placed
+     * @relatesalso MenuItem
+     */
+    void placeCursorAtEnd(MenuItem* item);
+
+   public:
+    /**
+     * Time when the toast started showing in milliseconds
+     */
+    unsigned long startTime;
+    /**
+     * How long the toast should Last in milliseconds
+     */
+    unsigned int duration;
+    /**
+     * LCD Display
+     */
+    
+#ifdef LCD_NO_I2C
+    LiquidCrystal* lcd;
+#else
+    LiquidCrystal_I2C* lcd;
+#endif
+    /**
+     * Constructor for the LcdMenu class
+     *
+     * @param maxRows rows on lcd display e.g. 4
+     * @param maxCols columns on lcd display e.g. 20
+     * @return new `LcdMenu` object
+     */
+    LcdMenu(uint8_t maxRows, uint8_t maxCols);
+    /**
+     * Call this function in `setup()` to initialize the LCD and the custom
+     * characters used as up and down arrows
+     *
+     * @param lcd_Addr address of the LCD on the I2C bus (default 0x27)
+     * @param menu menu to display
+     */
+    void setupLcdWithMenu(     
+#ifdef LCD_NO_I2C

```

Comment:
Could you instead place this function's definition in the `LcMenu.cpp` file, the header should contain only function declaration. You can see that the other functions have just been declared.